Fails to build due to conflicts caused by type redefinitions

With glibc `2.43+r5+g856c426a7534-1` the build fails with the following errors

```
[  1%] Building C object CMakeFiles/glcpp.dir/src/compiler/glsl/glcpp/glcpp-parse.c.o
In file included from /home/mario/Downloads/glsl-optimizer/include/c11/threads.h:66,
                 from /home/mario/Downloads/glsl-optimizer/src/mesa/main/mtypes.h:39,
                 from src/compiler/glsl/glcpp/glcpp-parse.y:32:
/home/mario/Downloads/glsl-optimizer/include/c11/threads_posix.h:54:9: warning: ‘ONCE_FLAG_INIT’ redefined
   54 | #define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
      |         ^~~~~~~~~~~~~~
In file included from /usr/include/stdlib.h:1191,
                 from src/compiler/glsl/glcpp/glcpp-parse.y:26:
/usr/include/bits/types/once_flag.h:25:9: note: this is the location of the previous definition
   25 | #define ONCE_FLAG_INIT __ONCE_FLAG_INIT
      |         ^~~~~~~~~~~~~~
/home/mario/Downloads/glsl-optimizer/include/c11/threads_posix.h:69:25: error: conflicting types for ‘once_flag’; have ‘pthread_once_t’ {aka ‘int’}
   69 | typedef pthread_once_t  once_flag;
      |                         ^~~~~~~~~
/usr/include/bits/types/once_flag.h:24:21: note: previous declaration of ‘once_flag’ with type ‘once_flag’
   24 | typedef __once_flag once_flag;
      |                     ^~~~~~~~~
/home/mario/Downloads/glsl-optimizer/include/c11/threads_posix.h:94:1: error: conflicting types for ‘call_once’; have ‘void(int *, void (*)(void))’
   94 | call_once(once_flag *flag, void (*func)(void))
      | ^~~~~~~~~
/usr/include/stdlib.h:1195:13: note: previous declaration of ‘call_once’ with type ‘void(once_flag *, void (*)(void))’
 1195 | extern void call_once (once_flag *__flag, void (*__func)(void));
      |             ^~~~~~~~~
make[2]: *** [CMakeFiles/glcpp.dir/build.make:93: CMakeFiles/glcpp.dir/src/compiler/glsl/glcpp/glcpp-parse.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/glcpp.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

After surrounding the definitions of `call_once` and `once_flag` with an `#ifndef __once_flag_defined` it builds successfully, although I am not sure this is the right fix.

--- a/include/c11/threads_posix.h
+++ b/include/c11/threads_posix.h
@@ -51,7 +51,9 @@ Configuration macro:
 #include <pthread.h>
 
 /*---------------------------- macros ----------------------------*/
+#ifndef __once_flag_defined
 #define ONCE_FLAG_INIT PTHREAD_ONCE_INIT
+#endif
 #ifdef INIT_ONCE_STATIC_INIT
 #define TSS_DTOR_ITERATIONS PTHREAD_DESTRUCTOR_ITERATIONS
 #else
@@ -66,7 +68,9 @@ typedef pthread_cond_t  cnd_t;
 typedef pthread_t       thrd_t;
 typedef pthread_key_t   tss_t;
 typedef pthread_mutex_t mtx_t;
+#ifndef __once_flag_defined
 typedef pthread_once_t  once_flag;
+#endif
 
 
 /*
@@ -90,11 +94,13 @@ impl_thrd_routine(void *p)
 
 /*--------------- 7.25.2 Initialization functions ---------------*/
 // 7.25.2.1
+#ifndef __once_flag_defined
 static inline void
 call_once(once_flag *flag, void (*func)(void))
 {
     pthread_once(flag, func);
 }
+#endif
 
 
 /*------------- 7.25.3 Condition variable functions -------------*/

